src/Storages/ObjectStorage/DataLakes/IStorageDataLake.h

```diff
@@ -102,12 +115,28 @@ class IStorageDataLake final : public StorageObjectStorage
         Storage::updateConfiguration(local_context);
 
         auto new_metadata = DataLakeMetadata::create(Storage::object_storage, base_configuration, local_context);
-        if (current_metadata && *current_metadata == *new_metadata)
-            return;
+        if (!current_metadata || (*current_metadata != *new_metadata))
+        {
+            if constexpr (std::is_same_v<IcebergMetadata, DataLakeMetadata>)
```

Comment:
We can have single method for all 3 places: `supportsDynamicMetadata`. Also need a comment about why we should not have different metadata here (because we should've already updated it in updateExternalDynamicMetadata)